common/mqtt_client/tests/mqtt_pub_sub_test.rs

```diff
@@ -24,17 +24,21 @@ fn sending_and_receiving_a_message() {
         )
         .await?;
         let _pkid = publisher.publish(message).await?;
+        let sleep = time::sleep(Duration::from_secs(1));
+        tokio::pin!(sleep);
 
-        tokio::select! {
-            msg = received.next() => Ok(msg),
-            _ = sleep(Duration::from_millis(1000)) => Ok(None)
+        loop {
+            tokio::select! {
+                msg = received.next() => { mqtt_server_handle.abort(); return Ok(msg);},
+                _ = &mut sleep => {mqtt_server_handle.abort(); return Ok(None);}
+            }
```

Comment:
* What problem do you try to solve?
* Why a loop with a return after the first iteration in any cases?
* What is the difference with the sleep started in the select?

Reply:
Yes, I tried fixing the issue by correct way of creating the timer inside the select statement. As Lukasz mentioned it worked for me for 2/3 tries. I tried the same test with mosquito broker, tests work properly even if I run them for hours continuously.

As per as creation of the sleep and pinning outside the select loop is required when the sleep is used multiple times.
Can find more about it here https://docs.rs/tokio/1.10.0/tokio/time/struct.Sleep.html